src/components/ReplaceServerDetails.js

```diff
@@ -0,0 +1,321 @@
+        </div>
+      </div>
+    );
+  }
+
+  renderDetailsTable() {
+    let rows = [];
+    let sections = [
+      {'role': this.data.role, 'ip-addr': this.data['ip-addr']},
+      {'server-group': this.data['server-group'], 'nic-mapping': this.data['nic-mapping']},
+      {'mac-addr': this.data['mac-addr'], 'ilo-ip': this.data['ilo-ip']},
+      {'ilo-user': this.data['ilo-user'], 'ilo-password': this.data['ilo-password']},
+    ];
+    sections.forEach((section, idx) => {
+      let keys = Object.keys(section);
+      rows.push(
+        <tr key={idx}>
+          <th>{translate('replace.server.details.' + keys[0])}</th>
+          <td>{keys[0].indexOf('password') !== -1  ? maskPassword(section[keys[0]]) : section[keys[0]]}</td>
+          <th>{translate('replace.server.details.' + keys[1])}</th>
+          <td>{keys[1].indexOf('password') !== -1 ? maskPassword(section[keys[1]]) : section[keys[1]]}</td>
+        </tr>);
+    });
+    return (
+      <table className='table table-condensed'>
+        <tbody>{rows}</tbody>
+      </table>
+    );
+  }
+
+  renderServerContent() {
+    return (
+      <div>
+        <div className='server-details-container'>
+          {this.renderDetailsTable()}
+        </div>
+        <div className='message-line'>{translate('server.replace.details.message')}</div>
+
+        <div className='server-details-container'>
+          {this.availableServerIds && this.availableServerIds.length > 0 && this.renderDropDown()}
+          {this.renderInput('mac-addr', 'text', true, 'server.mac.prompt', MacAddressValidator)}
+          {this.renderInput('ilo-ip', 'text', true, 'server.ipmi.ip.prompt',IpV4AddressValidator)}
+          {this.renderInput('ilo-user', 'text', true, 'server.ipmi.username.prompt')}
+          {this.renderInput('ilo-password', 'password', true, 'server.ipmi.password.prompt')}
+        </div>
+        <div className='server-details-container'>
+          <input className='replace-options' type='checkbox' value='installos'
+            checked={this.state.isInstallOsChecked} onChange={this.handleInstallOsCheck}/>
+          {translate('common.installos')}
+          {this.state.isInstallOsChecked &&
+            this.renderText('server.user.prompt', this.state.osInstallUsername)}
+          {this.state.isInstallOsChecked && this.renderOsPasswordInput()}
+        </div>
+        <div className='server-details-container'>
+          <input className='replace-options' type='checkbox' value='wipedisk'
+            checked={this.state.isWipeDiskChecked} onChange={this.handleWipeDiskCheck}/>
+          {translate('common.wipedisk')}
+        </div>
+      </div>
+    );
+  }
+
+  renderFooter() {
+    return (
+      <div className='btn-row input-button-container'>
+        <ActionButton type='default'
+          clickAction={this.handleCancel} displayLabel={translate('cancel')}/>
+        <ActionButton
+          isDisabled={!this.state.isFormValid}
```

Comment:
Replace the reference to `this.state.isFormValid` with a call to `this.isFormTextInputValid`

Reply:
k